### DISCUSSION TITLE
Regarding configmap

### DISCUSSION THREAD

**Discussion by afzal442** [iximiuz/client-go-examples#1] [Q&A]
I tinkered with the code I wondered why you have created configmap. Can we call it without configmap? Thanks.

e.g.
https://github.com/iximiuz/client-go-examples/blob/fe6ecf36b9fec883caefdd01895bc4640a75cb94/informer-typed-simple/main.go#L118

cc: @iximiuz

**Comment by iximiuz** [comment]
Hi Afzal! I use ConfigMaps there as a simple Kubernetes Object example. I could have chosen any other type of Kubernetes Objects (Deployments, Pods, etc.). I.e., no ConfigMaps is needed to make informers work. But w/o created/deleting objects the informer example would be incomplete.

**Reply by sbin64** [reply]
Hmm, makes sense now. I understood the reason behind the use of ConfigMaps obj. If we use any other K8s obj like Pods, Deployment etc., then I think we don't have to write other methods like creating/deleting objects. Isn't that correct? thanks.

**Reply by iximiuz** [reply]
I'm not sure if I understood your clarification fully. What exactly methods are you referring to?

**Reply by afzal442** [reply]
Sorry for that confusion. I mean that as we write methods for creating/deleting configmap object, we don't have to write methods when we consider other obj such as Pods, Deployments etc. Does that make sense? Thanks.

**Reply by iximiuz** [reply]
Since this is an [example of a typed informer](https://github.com/iximiuz/client-go-examples/blob/fe6ecf36b9fec883caefdd01895bc4640a75cb94/informer-typed-simple/main.go#L51), I don't think you can reuse it for other types of resources such as Pods or Deployments. But the code for these types of resources will be very similar.

**Reply by afzal442** [reply]
Thanks Ivan, got that. Another general confusion is whether we need to write methods for other resources like Pods etc. if we try to take an example of Pod object instead. e.g. `func createPod` and `func deletePod` as an use case of `informer()`. 🏷️
